### PR TITLE
Visitor can see articles in a specific category

### PR DESCRIPTION
**Pull request by arunbhalli**
**PT-Story:**  https://www.pivotaltracker.com/story/show/178315371
### User Story 
``` 
As a user
To get content that interests me most
I want to be able to access articles in a certain category
``` 
## Changes proposed in this pull request: 
* display article category on single article view
* onClick redirect to this category list View 
* create GET request to api/articles/category
* display message if there is no articles in that category
## What was learned:
*  how to refactor code into separate modules*  
___
## Screenshots (Native app) 

https://user-images.githubusercontent.com/57390886/119887725-c5a40f00-bf34-11eb-9f1c-ecda8a0aa4c4.mp4

### CODE REVIEW

**Comment by 1CIM** [comment]
So is it a WIP or not?

**Comment by DavDan1** [comment]
Not anymore  @1CIM  👍🏼

File: App.jsx [hunk 1]
```diff
@@ -37,8 +38,22 @@ const App = () => {
...
+          options={() => ({
+            headerBackTitle: 'Back',
+            title: 'Back',
+            headerStyle: {
+              backgroundColor: '#111518',
+            },
+            headerTintColor: '#CEC269',
+            headerTitleStyle: {
+              fontWeight: 'bold',
+            },
```
Comment (LucasKnudsen) [inline App.jsx:56]: We can refactor all the styling into a `screenOptions` as an attribute of the `Stack.Navigator`. 🚀

File: cypress/integration/userCanSeeArticlesInSpecificCategory.feature.js [hunk 2]
```diff
@@ -1,29 +1,58 @@
...
+    beforeEach(() => {
+      cy.intercept(
+        'GET',
+        'https://fakest-newzz.herokuapp.com/api/articles/Science',
+        {
+          articles: [],
+        }
+      );
+      cy.viewport('iphone-x');
+      cy.visit('/');
```
Comment (LucasKnudsen) [inline cypress/integration/userCanSeeArticlesInSpecificCategory.feature.js:47]: Do we have to specify viewport everytime or is it enough with once in the top beforeEach?

File: views/ViewByCategory.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  return (
+    <View style={styles.container}>
+      {noArticlesMessage ? (
+        <Text testID='no-articles-message' style={styles.errorMessage}>
+          No articles available at this moment
+        </Text>
+      ) : (
+        <FlatList
+          testID='view-by-category'
+          data={articles}
```
Comment (LucasKnudsen) [inline views/ViewByCategory.jsx:29]: We could perhaps also display the actual category as a header? :)

**Review by lukeperrin10 — COMMENTED** [review]
Good work, one suggestion, maybe add the vscode file to standardize the formatting, saw there was a few files that got a few double quotes changed in to singles.

**Comment by christerforsgren91** [comment]
Should we not merge this to the development branch? and there is a side-menu branch in the craft academy repo?

**Review by LucasKnudsen — COMMENTED** [review]
Looking neat boiz! 
I think I probably would've worked on menu functionality before adding a category feature, but sounded like that's next on the list? :P

File: App.jsx (hunk 1 above)
Comment (DavDan1) [reply App.jsx:56]: 👍🏼

File: cypress/integration/userCanSeeArticlesInSpecificCategory.feature.js (hunk 2 above)
Comment (DavDan1) [reply cypress/integration/userCanSeeArticlesInSpecificCategory.feature.js:47]: Yes you're right I think :)

**Review by tochman — APPROVED** [review]
Good job

**Pull request merged**